### PR TITLE
Make extensions not experimental for Platform 0.13 and above

### PR DESCRIPTION
**Pull request by prashantrewar**
#### Release notes

---

### Related

Resolves #___
#1287 
---

### CODE REVIEW

File: acceptance/detector_test.go [hunk 1]
```diff
@@ -349,6 +349,11 @@ fail: fail_detect_buildpack@some_version
 			os.RemoveAll(copyDir)
 		})
 
+		experimentalMode := "warn"
+		if api.MustParse(platformAPI).AtLeast("0.13") {
```
Comment (natalieparellano) [inline acceptance/detector_test.go:353]: Ah, for the detector tests we only exercise the "latest" Platform API version, hence `undefined: platformAPI` - we can just set the var to error always

File: acceptance/exporter_test.go [hunk 2]
```diff
@@ -141,11 +141,16 @@ func testExporterFunc(platformAPI string) func(t *testing.T, when spec.G, it spe
 						runImageFixtureTopLayerSHA := layers[len(layers)-1]
 						runImageFixtureSHA := inspect.ID
 
+						experimentalMode := "warn"
+						if api.MustParse(platformAPI).AtLeast("0.13") {
+							experimentalMode := "error"
```
Comment (natalieparellano) [inline acceptance/exporter_test.go:146]:
```suggestion
							experimentalMode = "error"
```

File: acceptance/exporter_test.go [hunk 3]
```diff
@@ -411,10 +416,15 @@ func testExporterFunc(platformAPI string) func(t *testing.T, when spec.G, it spe
 						runImageFixtureSHA, err := remoteImage.Digest()
 						h.AssertNil(t, err)
 
+						experimentalMode := "warn"
+						if api.MustParse(platformAPI).AtLeast("0.13") {
+							experimentalMode := "error"
```
Comment (natalieparellano) [inline acceptance/exporter_test.go:421]:
```suggestion
							experimentalMode = "error"
```

**Review by natalieparellano — COMMENTED** [review]
Nice work @prashantrewar! I think if you make the changes I suggested, the linter will be happy

File: acceptance/detector_test.go (hunk 1 above)
Comment (prashantrewar) [reply acceptance/detector_test.go]: Done!

File: acceptance/exporter_test.go (hunk 2 above)
Comment (prashantrewar) [reply acceptance/exporter_test.go]: Done!

File: acceptance/exporter_test.go (hunk 3 above)
Comment (prashantrewar) [reply acceptance/exporter_test.go]: Done!

**Comment by prashantrewar** [comment]
> Nice work @prashantrewar! I think if you make the changes I suggested, the linter will be happy

@natalieparellano, I updated the changes but the tests are still failing.

**Comment by natalieparellano** [comment]
Ah, so looking at the failure [here](https://github.com/buildpacks/lifecycle/actions/runs/7883240352/job/21509819641?pr=1295#step:5:5619) I think we need to change the test expectation [here](https://github.com/buildpacks/lifecycle/blob/d4e8609f4984968dd3e18239471494ca797f6c7f/acceptance/detector_test.go#L374)

**Review by natalieparellano — APPROVED** [review]

**Comment by prashantrewar** [comment]
Thank you so much @natalieparellano for the guidance and help.

**Pull request merged**